webapp/src/js/controllers/messages.js

```diff
@@ -37,103 +37,84 @@ angular
+          MessageContacts.conversation(selectedId).then(conversation => {
             ctrl.updateSelectedMessage({ messages: conversation });
           });
         }
-      } else {
-        ctrl.messages = options.messages || [];
       }
-      setActionBarData();
     };
 
-    var updateConversations = function(options) {
-      options = options || {};
-      if (!options.changes) {
-        ctrl.loading = true;
+    const setConversations = (conversations=[], { merge=false }={}) => {
+      if (merge) {
+        MessageListUtils.removeDeleted(ctrl.conversations, conversations);
+        MessageListUtils.mergeUpdated(ctrl.conversations, conversations);
+        ctrl.setConversations(conversations);
+        updateSelected();
+      } else {
+        ctrl.setConversations(conversations);
       }
-      return MessageContacts.list().then(function(messages) {
-        options.messages = messages;
-        setMessages(options);
-        ctrl.loading = false;
-      });
+      updateActionBar();
     };
 
-    $scope.markConversationRead = function(docs) {
-      var ids = _.pluck(docs, '_id');
-      var conversation = _.find(ctrl.messages, function(message) {
-        return ids.indexOf(message.id) !== -1;
+    const updateConversations = ({ merge=false }={}) => {
+      return MessageContacts.list().then(conversations => {
+        setConversations(conversations, { merge });
+        ctrl.loading = false;
       });
-      if (conversation) {
-        conversation.read = true;
-      }
     };
 
+    // updateActionBar();
     updateConversations()
-      .then(function() {
+      .then(() => {
         if (
           !$state.params.id &&
-          ctrl.messages.length &&
+          ctrl.conversations.length &&
           !responsive.isMobile() &&
           $state.is('messages.detail')
         ) {
-          $timeout(function() {
-            var first = $('.inbox-items li').first();
-            var state = {
+          $timeout(() => {
+            const first = $('.inbox-items li').first();
+            const state = {
               id: first.attr('data-record-id'),
               type: first.attr('data-record-type'),
             };
             $state.go('messages.detail', state, { location: 'replace' });
           });
         }
       })
-      .catch(function(err) {
-        $log.error('Error fetching contact', err);
-      });
+      .catch(err =>$log.error('Error fetching contact', err));
 
-    var changeListener = Changes({
+    const changeListener = Changes({
       key: 'messages-list',
-      callback: function() {
-        updateConversations({ changes: true });
-      },
-      filter: function(change) {
-        if (ctrl.currentTab !== 'messages') {
-          return false;
-        }
-        return (
+      callback: () => updateConversations({ merge: true }),
+      filter: change => {
+        return ctrl.currentTab === 'messages' &&
```

Comment:
is a set of parenthesis required here to maintain parity?

Reply:
I don't think so but it's more readable so I've added them.

Reply:
`false && false || true` evaluates to true
`false && (false || true) evalutes to false`

so i do think the parenthesis are required here